src/main/java/graphql/schema/GraphQLFieldDefinition.java

```diff
@@ -185,11 +163,7 @@ public Builder deprecate(String deprecationReason) {
 
         public GraphQLFieldDefinition build() {
             if (dataFetcher == null) {
-                if (isField) {
-                    dataFetcher = new FieldDataFetcher(name);
-                } else {
-                    dataFetcher = new PropertyDataFetcher(name);
-                }
+                dataFetcher = new PropertyDataFetcher(name);
```

Comment:
Done

On Tue, Sep 13, 2016 at 3:00 PM, Dmitry Minkovsky <dminkovsky@gmail.com>
wrote:

> No default fetcher at the `GraphQLFieldDefinition.Builder` level sounds
> good to me too. I'll update the code and docs to reflect that.
>
> I do sort of wish there could be a good default, but the presence of the
> good default has in its own right [caused confusion](https://github.com/
> graphql-java/graphql-java/issues/192#issuecomment-242254224).
>
> вторник, 13 сентября 2016 г. пользователь Mark Philpot написал:
>
> In src/main/java/graphql/schema/GraphQLFieldDefinition.java
>> <https://github.com/graphql-java/graphql-java/pull/206#discussion_r78621307>
>> :
>>
>> > @@ -185,11 +163,7 @@ public Builder deprecate(String deprecationReason) {
>> >
>> >          public GraphQLFieldDefinition build() {
>> >              if (dataFetcher == null) {
>> > -                if (isField) {
>> > -                    dataFetcher = new FieldDataFetcher(name);
>> > -                } else {
>> > -                    dataFetcher = new PropertyDataFetcher(name);
>> > -                }
>> > +                dataFetcher = new PropertyDataFetcher(name);
>>
>> I personally use objects for the type safety, but I can understand a Map
>> might be a new user's first choice. In that light, would vote for being
>> explicit with no default since we are breaking out Map and Property
>> functionality from Field
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/graphql-java/graphql-java/pull/206/files/eb75d57f0bbcaed455812218fd81beda028a8379#r78621307>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AANWZVdcsJiMykODP9z4d0kA-28aWE-jks5qpvDtgaJpZM4J7Pgu>
>> .
>>
>